library/PlugLoad-posix.cpp

```diff
@@ -38,8 +38,12 @@ namespace DFHack
     {
         return (void *) dlsym((void *)plugin, function);
     }
-    void ClosePlugin (DFLibrary * plugin)
+    bool ClosePlugin (DFLibrary * plugin)
     {
-        dlclose((void *) plugin);
+        // dlclose returns 0 on success, nonzero on failure
+        int res = dlclose((void *) plugin);
+        if (res != 0)
+            WARN(plugins).print("dlclose failed with error code %x\n", res);
```

Comment:
needs a `DBG_DECLARE(core, plugins, DebugCategory::LINFO);` in this file

Reply:
yeah that's part of why i wanted to combine the two into a single file